internal/parse/parser.go

```diff
@@ -243,58 +236,65 @@ func (p *Parser) skipName() bool {
 //  - bool == false, err == nil
 //		The construct was not the one we are looking for
 
-// parseIdentifier parses either a name made up only of nameBytes or an
-// asterisk.
-func (p *Parser) parseIdentifier() (string, bool) {
-	if p.skipByte('*') {
-		return "*", true
-	}
+type asteriskFlag int
 
-	idStart := p.pos
+const (
+	allowAsterisk asteriskFlag = iota
+	disallowAsterisk
+)
+
+// parseIdentifier parses a name made up of only nameBytes, or alternatively a
+// single asterisk if `allowAsterisk` is passed. On success it returns the
+// parsed string and true. Otherwise, it returns the empty string and false.
+func (p *Parser) parseIdentifier(asteriskF asteriskFlag) (string, bool) {
+	if asteriskF == allowAsterisk {
+		if p.skipByte('*') {
+			return "*", true
+		}
+	}
+	mark := p.pos
 	if p.skipName() {
-		return p.input[idStart:p.pos], true
+		return p.input[mark:p.pos], true
 	}
 	return "", false
 }
 
-// parseFullName parses a column name or Go field name, optionally dot-prefixed by
-// its table or type name respectively.
-func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+// parseGoObject parses a source or target go object of the form Prefix.Name
+// where the type is the Prefix and the field is the Name (this applies to maps
+// and structs). On success it returns the parsed FullName, true and nil. If a
+// Go object is found, but not formatted correctly, false and an error are
+// returned. Otherwise the error is nil.
+func (p *Parser) parseGoObject() (FullName, bool, error) {
```

Comment:
This looks fine, but wasn't the point of having a parseFullName that "database.column" and "Type.db_field" look the same?

Reply:
Changed.